File: 11-ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int r = height.size() - 1, l = 0, mx = 0;
        while (l < r) {
            mx = max(mx, min(height[r], height[l]) * (r- l));
            if (height[l] < height[r]) l++;
            else r--;
        }
        return mx;
    }
};

File: 11-ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        mx = 0
        l = 0
        n = len(height)
        r = n - 1
        while l < r:
            area = min(height[r], height[l]) * (r-l)
            mx = max(mx, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        

        return mx
